app/models/comment.rb

```diff
@@ -1,4 +1,9 @@
 class Comment < ApplicationRecord
-  belongs_to :author_id
-  belongs_to :post_id
+  belongs_to :author, class_name: :User
+  belongs_to :post, class_name: :Post
+
+  def increase_comment_counter
+    post.comments_counter += 1
+    post.save
+  end
```

Comment:
- **[OPTIONAL]** I think it would be nice if you triggered this method using the `after_save` active record callback so that when a comment is saved Rails automatically calls this function. I also think we can refactor this code to be briefer.

  ```
  .
  .
  after_save :increase_comment_counter
  .
  .
  private

  def increase_comment_counter
    post.increment!(:comments_counter)
  end
  ```

Reply:
Hi @DammyShittu, I will implement the suggested changes. Thank you 🙏